R/stop-level_excel.R

```diff
@@ -1,128 +1,93 @@
-    # remove extraneous rows, keep only common name and stops
-    dplyr::select(common_name, dplyr::matches("s[0-9]+")) %>%
-    mutate(hist_xls_extract_county_num_year_from_filename(filename),
-      source = "prep_sld, hist xls",
-      route_num = as.integer(route_num)
-    ) %>%
-    mutate(across(s1:s20, ~ ifelse(is.na(.), 0, .))) %>% # change NAs to 0
-    # change x and X to 1, following Haven Wiley's example
-    purrr::map_dfr(., ~ ifelse(. == "x", 1, .)) %>%
-    purrr::map_dfr(., ~ ifelse(. == "X", 1, .)) %>%
-    # ensure counts are integers, not characters
-    mutate(across(s1:s20, as.integer)) %>%
-    hist_xls_add_species_and_route_info() %>%
-    # pivot
-    pivot_longer(
+process_stop_level_xls <- function(file) {
+  readxl::read_excel(
+    path = file,
+    col_types = rep("text", 25),
+    col_names = c(
+      "species_code", paste("s", 1:20, sep = ""),
+      "sum", "sequence", "species_code2", "common_name"
+    ),
+    skip = 5,
+    .name_repair = "unique_quiet"
+  ) |>
+    # Filters to rows where the *first* column is a species code
+    # (four capital letters OR acsp - "Accipiter sp.")
+    dplyr::filter_at(1, (\(x) stringr::str_detect(x, "^[A-Z]{4}$|acsp"))) |>
+    # Correct common names
+    dplyr::mutate(
+      common_name = str_replace_all(
+        common_name,
+        c(
+          "Rock Dove" = "Rock Pigeon",
+          "^Whip-poor-will$" = "Eastern Whip-poor-will"
+        )
+      )
+    ) |>
+    # Remove extraneous rows, keep only common name and stops
+    dplyr::select(common_name, dplyr::matches("s[0-9]+")) |>
+    # Pivot data from wide to long
+    tidyr::pivot_longer(
       cols = s1:s20,
-      names_to = "stop_num", names_prefix = "s", values_to = "count"
-    ) %>%
-    mutate(stop_num = as.integer(stop_num))
-
-  hist_xls_run_checks(hist_xls)
-}
-
-#' Filters to only rows where the first column is a species code
-#' that is four capital letters
-#' or acsp - "Accipiter sp."
-#' @importFrom stringr str_detect
-#' @importFrom dplyr filter
-#' @param x a dataframe from a historical .xls
-hist_xls_filter_to_species_code_rows <- \(x){
-  x %>%
-    filter(str_detect(.[[1]], "^[A-Z][A-Z][A-Z][A-Z]$|acsp"))
-}
-
-
-#' Rename the columns of a historical xls
-#' @importFrom dplyr %>%
-#' @param x a dataframe from a historical .xls
-hist_xls_rename_columns <- \(x){
-  x %>%
-    `colnames<-`(c(
-      "species_code", paste("s", 1:20, sep = ""), "sum",
-      "sequence", "species_code2", "common_name"
-    ))
-}
-
+      names_to = "stop_num",
+      names_prefix = "s",
+      values_to = "count"
+    ) |>
+    # Parse counts as integers
+    dplyr::mutate(
+      as_tibble(extract_info_from_filename(file)),
+      stop_num = as.integer(stop_num),
+      route = make_route_id(county, route_num),
+      count = as.integer(
+        case_when(
+          count %in% c("x", "X") ~ "1",
+          is.na(count) ~ "0",
+          TRUE ~ count
+        )
+      )
+    ) |>
```

Comment:
Need to taxonomy and observer info.

Reply:
Observer info can be added in with survey_list.csv rather that mbbs_survey_events (since the .rda is going away)

Reply:
But also I'm going to go edit the process_observers so it can be pulled from a survey_events.csv

Reply:
Well, never mind on that for now, as the mbbs.rda is also gone and I need that to generate survey_events. To add observer info leftjoin in survey_list rather than survey_events.

Reply:
I'm thinking taxonomy and observers can be updated in a downstream process so that all data source are handled consistently.